File: leetcode2875/leetcode2875.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int minSizeSubarray(vector<int>& nums, int target) {
            long long total = reduce(nums.begin(), nums.end(), 0LL);
            int n = nums.size();
            int ans = INT_MAX;
            int left = 0;
            long long sum = 0;
            for (int right = 0; right < n * 2; right++) {
                sum += nums[right % n];
                while (sum > target % total) {
                    sum -= nums[left++ % n];
                }
                if (sum == target % total) {
                    ans = min(ans, right - left + 1);
                }
            }
            return ans == INT_MAX ? -1 : ans + target / total * n;
        }
    };
    

File: leetcode2875/leetcode2875.py
class Solution:
    def minSizeSubarray(self, nums: List[int], target: int) -> int:
        total = sum(nums)
        n = len(nums)
        ans = inf
        left = s = 0
        for right in range(n * 2):
            s += nums[right % n]
            while s > target % total:
                s -= nums[left % n]
                left += 1
            if s == target % total:
                ans = min(ans, right - left + 1)
        return ans + target // total * n if ans < inf else -1

